fe/fe-core/src/main/java/com/starrocks/server/StorageVolumeMgr.java

```diff
@@ -15,144 +15,132 @@
-            // TODO: check ref count
-            StorageVolume sv = nameToSV.remove(name);
-            idToSV.remove(sv.getId());
-        }
-    }
+    public abstract void removeStorageVolume(String name) throws AnalysisException, DdlException;
 
-    public void updateStorageVolume(AlterStorageVolumeStmt stmt) throws AnalysisException {
+    public void updateStorageVolume(AlterStorageVolumeStmt stmt) throws AnalysisException, DdlException {
         Map<String, String> params = new HashMap<>();
         Optional<Boolean> enabled = parseProperties(stmt.getProperties(), params);
         updateStorageVolume(stmt.getName(), params, enabled, stmt.getComment());
     }
 
-    public void updateStorageVolume(String name, Map<String, String> params, Optional<Boolean> enabled, String comment)
-            throws AnalysisException {
-        try (LockCloseable lock = new LockCloseable(rwLock.writeLock())) {
-            Preconditions.checkState(nameToSV.containsKey(name),
-                    "Storage Volume '%s' does not exist", name);
+    public abstract void updateStorageVolume(String name, Map<String, String> params, Optional<Boolean> enabled, String comment)
+            throws AnalysisException, DdlException;
 
-            StorageVolume sv = nameToSV.get(name);
-            if (!params.isEmpty()) {
-                sv.setCloudConfiguration(params);
-            }
+    public void setDefaultStorageVolume(SetDefaultStorageVolumeStmt stmt) throws AnalysisException, DdlException {
+        setDefaultStorageVolume(stmt.getName());
+    }
 
-            if (enabled.isPresent()) {
-                if (!enabled.get()) {
-                    Preconditions.checkState(!name.equals(defaultSV), "Default volume can not be disabled");
-                }
-                sv.setEnabled(enabled.get());
-            }
+    public abstract void setDefaultStorageVolume(String svKey) throws AnalysisException, DdlException;
 
-            if (!comment.isEmpty()) {
-                sv.setComment(comment);
-            }
-        }
+    public long getDefaultStorageVolumeId() {
+        return defaultStorageVolumeId;
     }
 
-    public void setDefaultStorageVolume(SetDefaultStorageVolumeStmt stmt) {
-        setDefaultStorageVolume(stmt.getName());
-    }
+    public abstract boolean exists(String svKey) throws DdlException;
 
-    public void setDefaultStorageVolume(String svKey) {
-        try (LockCloseable lock = new LockCloseable(rwLock.writeLock())) {
-            Preconditions.checkState(nameToSV.containsKey(svKey),
-                    "Storage Volume '%s' does not exist", svKey);
-            StorageVolume sv = nameToSV.get(svKey);
-            sv.setIsDefault();
-            defaultSV = svKey;
-        }
-    }
+    public abstract StorageVolume getStorageVolumeByName(String svKey) throws AnalysisException;
+
+    public abstract StorageVolume getStorageVolume(long storageVolumeId) throws AnalysisException;
```

Comment:
need consider if this is a good practice to directly return StorageVolume object to caller who can modify StorageVolume as wish.